manifests/trusted_external_command.pp

```diff
@@ -25,6 +28,7 @@
   String $instance,
   String $user,
   String $password,
+  String $oauth_token,
```

Comment:
Hmm, this will fail if I specify `user/password` but _not_ `oauth_token` b/c Puppet will treat `oauth_token` as a required param (and similar scenario for when you specify `oauth_token` and `user/password`).

I guess we could just tell them "Hey, just leave `oauth_token` empty or `user/password` empty", but that seems bad. So, I think you may need to set defaults for `user`, `password`, and `oauth_token` then add some validation to the class ensuring that at least one set of credentials are specified.

Reply:
We should also add unit tests for that scenario (and credentials validation) to `spec/classes/<...>.pp`.

I also wonder if we should do a dry test in the module (like not only validate that credentials are specified, but also that they are correct). @bmjen Thoughts? I think it might lead to a better UX. Otherwise if you specify wrong credentials, then you will see that error in the `servicenow.rb` script instead of when applying the class.